tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/konan/KonanBuildingConfig.kt

```diff
@@ -143,31 +150,53 @@ abstract class KonanBuildingConfig<T: KonanBuildingTask>(private val name_: Stri
 
     internal operator fun get(target: KonanTarget) = targetToTask[target]
 
-    fun getByTarget(target: String) = findByTarget(target) ?: throw NoSuchElementException("No such target for artifact $name: ${target}")
+    fun getByTarget(target: String) =
+        findByTarget(target) ?: throw NoSuchElementException("No such target for artifact $name: $target")
+
     fun findByTarget(target: String) = this[project.hostManager.targetByName(target)]
 
-    fun getArtifactByTarget(target: String) = getByTarget(target).artifact
-    fun findArtifactByTarget(target: String) = findByTarget(target)?.artifact
+    fun getArtifactByTarget(target: String) = getByTarget(target).get().artifact
+    fun findArtifactByTarget(target: String) = findByTarget(target)?.get()?.artifact
 
     // Common building DSL.
 
-    override fun artifactName(name: String)  = forEach { it.artifactName(name) }
+    override fun artifactName(name: String) = tasks().forEach { it.configure { t -> t.artifactName(name) } }
+
+    fun baseDir(dir: Any) =
+        tasks().forEach {
+            it.configure { t ->
+                t.destinationDir(
+                    project.file(dir).targetSubdir(t.konanTarget)
+                )
+            }
+        }
+
+    override fun libraries(closure: Closure<Unit>) =
+        tasks().forEach { it.configure { t -> t.libraries(closure) } }
```

Comment:
(minor) The construction `tasks().forEach { it.configure { ... } }` is very common now. May be move it to a separate function? E.g.

```kotlin
fun configureTasks(block: (T) -> Unit) = tasks().forEach { it.configure(block) }
```

Reply:
I'll do that inside Kotlin repo with improvements to the test infra (making it configure lazily also)